votor/src/certificate_pool.rs

```diff
@@ -183,7 +186,7 @@ impl CertificatePool {
         transaction: &VoteMessage,
         validator_vote_key: &Pubkey,
         validator_stake: Stake,
-    ) -> bool {
+    ) -> Option<Stake> {
```

Comment:
I think grouping the return value into an `entry_stake` can be a bit confusing in some cases, i.e. for skip votes it's the total_stake which is a bit confusing, whereas for the notarization votes its the entry stake

Maybe here we can return an 

```
enum NewVote {
      Skip(Stake),
      Notarize(Slot, BlockId, Stake),
}
```

And then match on this result to either call `add_notarize` or `add_skip` once we break up ` add_vote` https://github.com/anza-xyz/alpenglow/blob/69a256e12d885287cf3dbeef9dea64d250bad7d6/votor/src/certificate_pool/slot_stake_counters.rs#L25

Reply:
@wen-coding hmm yeah, seems like we can't avoid passing the `vote` either, feel free to revert back to the entry_stake and the separate vote, sorry for the randomizaiton

Reply:
No problem. If in the future we move more stuff into SlotStakeCounters we can always refactor this into an enum again.